src/lib/es2015.promise.d.ts

```diff
@@ -3,47 +3,76 @@
  */
 interface Promise<T> {
     /**
+     * Creates a new Promise with the same internal state of this Promise.
+     * @returns A Promise.
+     */
+    then(): Promise<T>;

```

Comment:
I thought overloads should be arranged longest-to-shortest to make sure that the most restrictive overloads matches first. Why change that here?